packages/analytics-browser-test/test/web-attribution.test.ts

```diff
@@ -0,0 +1,603 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import * as amplitude from '@amplitude/analytics-browser';
+import { UUID } from '@amplitude/analytics-core';
+import { default as nock } from 'nock';
+import { path, url as httpEndPoint } from './constants';
+import { success } from './responses';
+import 'isomorphic-fetch';
+import {
+  generateEvent,
+  generateAttributionEvent,
+  generateSessionStartEvent,
+  generateSessionEndEvent,
+  generatePageViewEvent,
+  navigateTo,
+} from './helpers';
+
+describe('Web attribution', () => {
+  const defaultTracking = {
+    attribution: true,
+    fileDownloads: false,
+    formInteractions: false,
+    pageViews: true,
+    sessions: true,
+  };
+
+  let apiKey = '';
+  let client = amplitude.createInstance();
+
+  const event_upload_time = '2023-01-01T12:00:00:000Z';
+  Date.prototype.toISOString = jest.fn(() => event_upload_time);
+
+  beforeEach(() => {
+    client = amplitude.createInstance();
+    apiKey = UUID();
+  });
+
+  describe('Page load', () => {
+    describe('in a new session', () => {
+      let eventId = -1;
+      beforeEach(() => {
+        eventId = -1;
+      });
+
+      afterEach(() => {
+        cleanup();
+      });
+
+      test('should track all UTMs and referrers', async () => {
+        const url = 'https://www.example.com?utm_source=test_utm_source';
+        navigateTo(url);
```

Comment:
Should we also pass a referral here to make sure we test that referral is also tracked?

Reply:
Added.